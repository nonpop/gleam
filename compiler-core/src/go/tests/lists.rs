use crate::assert_go;

#[test]
fn list_literals() {
    assert_go!(
        r#"
fn go(x) {
    []
    [1]
    [1, 2]
    [1, 2, ..x]
}
"#,
    );
}

#[test]
fn long_list_literals() {
    assert_go!(
        r#"
fn go() {
  ["111111111111111111111111111111111111111111111111111111111111111111111111"]
  ["11111111111111111111111111111111111111111111", "1111111111111111111111111111111111111111111"]
}
"#,
    );
}

#[test]
fn multi_line_list_literals() {
    assert_go!(
        r#"
fn go(x) {
    [{True 1}]
}
"#,
    );
}

#[test]
fn list_constants() {
    assert_go!(
        r#"
const a = []
const b = [1, 2, 3]
"#,
    );
}

#[test]
fn list_destructuring() {
    assert_go!(
        r#"
fn go(x, y) {
  let assert [] = x
  let assert [a] = x
  let assert [1, 2] = x
  let assert [_, #(3, b)] = y
  let assert [head, ..tail] = y
}
"#,
    );
}

#[test]
fn equality() {
    assert_go!(
        r#"
fn go() {
  [] == [1]
  [] != [1]
}
"#,
    );
}

#[test]
fn case() {
    assert_go!(
        r#"
fn go(xs) {
  case xs {
    [] -> 0
    [_] -> 1
    [_, _] -> 2
    _ -> 9999
  }
}
"#,
    );
}

// https://github.com/gleam-lang/gleam/issues/2904
#[test]
fn tight_empty_list() {
    assert_go!(
        r#"
fn go(func) {
  let huuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuge_variable = []
}
"#,
    );
}
