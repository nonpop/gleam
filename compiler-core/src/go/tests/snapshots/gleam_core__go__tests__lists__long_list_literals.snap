---
source: compiler-core/src/go/tests/lists.rs
expression: "\nfn go() {\n  [\"111111111111111111111111111111111111111111111111111111111111111111111111\"]\n  [\"11111111111111111111111111111111111111111111\", \"1111111111111111111111111111111111111111111\"]\n}\n"
snapshot_kind: text
---
----- SOURCE CODE

fn go() {
  ["111111111111111111111111111111111111111111111111111111111111111111111111"]
  ["11111111111111111111111111111111111111111111", "1111111111111111111111111111111111111111111"]
}


----- COMPILED GO
package mod_P

import (
  gleam_P "example.com/todo/gleam"
)

const _ = gleam_P.Use_Import
const Use_Import byte = 0

func go π() gleam_P.List_t[gleam_P.String_t] {
  _ = gleam_P.ToList[gleam_P.String_t](
    "111111111111111111111111111111111111111111111111111111111111111111111111",
  )
  return gleam_P.ToList[gleam_P.String_t](
    "11111111111111111111111111111111111111111111",
    "1111111111111111111111111111111111111111111",
  )
}
